### DISCUSSION TITLE
librdkafka has no truststore parameter

### DISCUSSION THREAD

**Discussion by kartikzheng** [fluent/fluent-bit#6229] [Q&A] [answered]
fluent-bit logs are output to authenticated kafka, which uses ssl authentication, but librdkafka has no truststore parameter. What should I do?

Configuration for kafka:
```
listeners=SSL://XXXX:9092
ssl.keystore.location=/data/kafka/ssl/server.keystore.jks
ssl.keystore.password=XXXX
ssl.key.password=XXXX
ssl.truststore.location=/data/kafka/ssl/server.truststore.jks
ssl.truststore.password=XXXX
ssl.endpoint.identification.algorithm=
security.inter.broker.protocol=SSL
```

**Comment by OneCricketeer** [comment, marked as answer]
Have you seen this? https://stackoverflow.com/questions/56397039/librdkafka-consumer-and-ssl-configuration

**Reply by kartikzheng** [reply]
Thank you for your answer, I have done the authentication with keystore, but how to access the ssl authentication kafka by fluent-bit, there seems to be no official example, can not find the truststore parameter.

```
[OUTPUT]
    Name        kafka
    Match       *
    Brokers     192.168.1.3:9092
    Topics      test
    Schema_str  {"name":"avro_logging","type":"record","fields":[{"name":"timestamp","type":"string"},{"name":"stream","type":"string"},{"name":"log","type":"string"},{"name":"kubernetes","type":{"name":"krec","type":"record","fields":[{"name":"pod_name","type":"string"},{"name":"namespace_name","type":"string"},{"name":"pod_id","type":"string"},{"name":"labels","type":{"type":"map","values":"string"}},{"name":"annotations","type":{"type":"map","values":"string"}},{"name":"host","type":"string"},{"name":"container_name","type":"string"},{"name":"docker_id","type":"string"},{"name":"container_hash","type":"string"},{"name":"container_image","type":"string"}]}},{"name":"cluster_name","type":"string"},{"name":"fabric","type":"string"}]}
    Schema_id some_schema_id
    rdkafka.client.id some_client_id
    rdkafka.debug All
    rdkafka.enable.ssl.certificate.verification true

    rdkafka.ssl.certificate.location /certs/some.cert
    rdkafka.ssl.key.location /certs/some.key
    rdkafka.ssl.ca.location /certs/some-bundle.crt
    rdkafka.security.protocol ssl
    rdkafka.request.required.acks 1
    rdkafka.log.connection.close false

    Format avro
    rdkafka.log_level 7
    rdkafka.metadata.broker.list 192.168.1.3:9092
```

**Reply by OneCricketeer** [reply]
> can not find the truststore parameter.

Because there isn't one... Truststore is only a JVM concept only, not used by OpenSSL

**Reply by kartikzheng** [reply]
Thank you. How does fluent-bit access ssl authenticated kafka

**Reply by OneCricketeer** [reply]
It relies on librdkafka, as you've already discovered

https://github.com/edenhill/librdkafka/wiki/Using-SSL-with-librdkafka

**Reply by kartikzheng** [reply]
Thank you. I'll have a try

**Reply by kartikzheng** [reply]
Have you had this problem? thank you
`[error] [output:kafka:kafka.0] fluent-bit#producer-1: [thrd:app]: fluent-bit#producer-1: ssl://XXX:9092/bootstrap: SSL handshake failed: ssl/record/ssl3_record.c:332: error:1408F10B:SSL routines:ssl3_get_record:wrong version number (after 3ms in state SSL_HANDSHAKE, 30 identical error(s) suppressed)`
